src/main/java/seedu/address/storage/JsonAdaptedPatient.java

```diff
@@ -0,0 +1,116 @@
+        this.status = status;
+        this.remark = remark;
+    }
+
+    /**
+     * Converts a given {@code Patient} into this class for Jackson use.
+     */
+    public JsonAdaptedPatient(Patient source) {
+        super(source);
+        this.height = source.getHeight().height;
+        this.weight = source.getWeight().weight;
+        this.diagnosis = source.getDiagnosis().diagnosis;
+        this.status = source.getStatus().toString();
+        this.remark = source.getRemark().toString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted patient object into the model's {@code Patient} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted doctor.
+     */
+    public Patient toModelType() throws IllegalValueException {
+        if (height == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Height.class.getSimpleName()));
+        }
+        if (!Height.isValidHeight(height)) {
+            throw new IllegalValueException(Height.MESSAGE_CONSTRAINTS);
+        }
+        final Height modelHeight = new Height(height);
+
+        if (weight == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Weight.class.getSimpleName()));
+        }
+        if (!Weight.isValidWeight(weight)) {
+            throw new IllegalValueException(Weight.MESSAGE_CONSTRAINTS);
+        }
+        final Weight modelWeight = new Weight(weight);
+
+        if (diagnosis == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Diagnosis.class.getSimpleName()));
+        }
+        if (!Diagnosis.isValidDiagnosis(diagnosis)) {
+            throw new IllegalValueException(Diagnosis.MESSAGE_CONSTRAINTS);
+        }
+        final Diagnosis modelDiagnosis = new Diagnosis(diagnosis);
+
+        if (status == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Status.class.getSimpleName()));
+        }
+        if (!Status.isValidStatus(status)) {
+            throw new IllegalValueException(Status.MESSAGE_CONSTRAINTS);
+        }
+        final Status modelStatus = new Status(status);
+
+        if (remark == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Remark.class.getSimpleName()));
+        }
+        if (!Remark.isValidRemark(remark)) {
+            throw new IllegalValueException(Remark.MESSAGE_CONSTRAINTS);
+        }
+        final Remark modelRemark = new Remark(remark);
+
+        Person doctorPerson = super.toModelType();
+        return new Patient(doctorPerson.getName(), doctorPerson.getPhone(),
+                doctorPerson.getEmail(), modelHeight, modelWeight,
+                modelDiagnosis, modelStatus, modelRemark, doctorPerson.getTags());
```

Comment:
`doctorPerson` should be `patientPerson`

Reply:
Change made!